### PR TITLE
[cli] add docs for `partitionid` through `pollperiod`

### PR DESCRIPTION
**Pull request by kylorene**

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -4229,62 +4229,50 @@ template <> otError Interpreter::Process<Cmd("leaderdata")>(Arg aArgs[])
...
 
+    /**
+     * @cli partitionid
+     * @code
+     * partitionid
+     * 4294967295
+     * Done
+     * @endcode
+     * @par api_copy
+     * #otThreadGetPartitionId
```
Comment (Vyrastas) [inline src/cli/cli.cpp:4245]: Switch the api_copy to an `@sa` here

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -4229,62 +4229,50 @@ template <> otError Interpreter::Process<Cmd("leaderdata")>(Arg aArgs[])
...
+     * 4294967295
+     * Done
+     * @endcode
+     * @code
+     * partitionid preferred 0xffffffff
+     * Done
+     * @endcode
+     * @cparam partitionid preferred @ca{partitionid}
+     * @sa otThreadGetPreferredLeaderPartitionId
+     *@sa otThreadSetPreferredLeaderPartitionId
```
Comment (Vyrastas) [inline src/cli/cli.cpp:4268]: This imported okay, but for readability need a space between `*` and `@sa`

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -5896,6 +5883,7 @@ void Interpreter::HandlePingStatistics(const otPingSenderStatistics *aStatistics
  * The address can be an IPv4 address, which will be synthesized to an IPv6 address using the preferred NAT64 prefix from the network data.
  * @par
  * Note: The command will return InvalidState when the preferred NAT64 prefix is unavailable.
+  * @sa otPingSenderPing
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5886]: Needs to move left one space

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -4234,12 +4234,41 @@ template <> otError Interpreter::Process<Cmd("partitionid")>(Arg aArgs[])
...
+     * Get the Thread Network Partition ID.
+     */
     if (aArgs[0].IsEmpty())
     {
         OutputLine("%lu", ToUlong(otThreadGetPartitionId(GetInstancePtr())));
         error = OT_ERROR_NONE;
     }
 #if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    /**
+     * @cli partitionid preferred (partitionid) (get,set)
```
Comment (Vyrastas) [inline src/cli/cli.cpp:4256]: Let's remove the (partitionid) from the name

`* @cli partitionid preferred (get,set)`

File: src/cli/cli.cpp [hunk 5]
```diff
@@ -4234,12 +4234,41 @@ template <> otError Interpreter::Process<Cmd("partitionid")>(Arg aArgs[])
...
+     * @endcode
+     * @code
+     * partitionid preferred 0xffffffff
+     * Done
+     * @endcode
+     * @cparam partitionid preferred @ca{partitionid}
+     * @sa otThreadGetPreferredLeaderPartitionId
+     * @sa otThreadSetPreferredLeaderPartitionId
+     * @par
+     * Note: OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE is required.
```
Comment (Vyrastas) [inline src/cli/cli.cpp:4270]: This doesn't need to be a note for our docs, but put the flag in code font

`OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is required.

File: src/cli/cli.cpp [hunk 6]
```diff
@@ -5928,13 +5999,39 @@ template <> otError Interpreter::Process<Cmd("ping")>(Arg aArgs[])
...
+ */
 template <> otError Interpreter::Process<Cmd("platform")>(Arg aArgs[])
 {
     OT_UNUSED_VARIABLE(aArgs);
     OutputLine("%s", OPENTHREAD_CONFIG_PLATFORM_INFO);
     return OT_ERROR_NONE;
 }
 
+/**
+ * @cli pollperiod (pollperiod)
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6020]: `* @cli pollperiod (get,set)`

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli.cpp [hunk 7]
```diff
@@ -4267,7 +4266,7 @@ template <> otError Interpreter::Process<Cmd("partitionid")>(Arg aArgs[])
      * @sa otThreadGetPreferredLeaderPartitionId
      * @sa otThreadSetPreferredLeaderPartitionId
      * @par
-     * Note: OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE is required.
+     * 'OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE' is required.
```
Comment (Vyrastas) [inline src/cli/cli.cpp:4269]: Make it backticks around OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE  so it's code font

**Review by Vyrastas — APPROVED** [review]
Please rebase/squash

**Review by jwhui — APPROVED** [review]

**Pull request merged**